client/ethereum.go

```diff
@@ -411,17 +428,32 @@ func (e *EthereumClient) subscribeToNewHeaders() error {
 		case err := <-subscription.Err():
 			return err
 		case header := <-headerChannel:
+			e.recordHeader(header)
 			e.receiveHeader(header)
 		case <-e.doneChan:
 			return nil
 		}
 	}
 }
 
+// recordHeader saves header either to first seen headers or to reorged/duplicated
+func (e *EthereumClient) recordHeader(header *types.Header) {
```

Comment:
I'm not a fan of always recording headers as this means memory usage will grow the longer the test runs/more blocks received. Can this be changed into a header subscription where the test will wait for a re-org and within that subscriber it records block headers?